src/App.js

```diff
@@ -37,35 +49,110 @@ function AppNavbar() {
             />{" "}
           </Link>
         </Navbar.Brand>
-        <PWAInstallButton />
+        <Dropdown className="dropdown--translation">
+          <Dropdown.Toggle as={IconToggle} />
+          <Dropdown.Menu>
+            {Object.keys(AVAILABLE_LANGUAGES).map(key => {
+              const langValue = AVAILABLE_LANGUAGES[key];
+              return (
+                <Dropdown.Item
+                  key={langValue}
+                  active={currentLanguage === langValue}
+                  onClick={() => setLanguage(langValue)}
+                >
+                  {langValue}
+                </Dropdown.Item>
+              );
+            })}
+          </Dropdown.Menu>
+        </Dropdown>
+        {/* <PWAInstallButton /> */}
       </Navbar>
     </>
   );
-}
+};
 
-function App() {
-  return (
-    <Router history={history}>
-      <div className="App">
-        <ScrollToTop />
-        <AppNavbar />
-        <div className="page">
-          <Switch>
-            <Route path="/update" component={SubmitPage} />
-            <Route path="/location" component={LocationSelectionPage} />
-            <Route path="/about" component={AboutPage} />
-            <Route path="/" component={Homepage} />
-          </Switch>
-        </div>
-        <footer className="m-0 p-0">
-          <div className="container py-4 text-center text-uppercase">
-            <Link to="/">Home</Link> · <Link to="/location">Add a store</Link> ·{" "}
-            <Link to="/about">About</Link>
+const AppContext = createContext();
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      language: this.getDefaultLanguage(),
+    };
+  }
+
+  setLanguage = language => {
+    this.persistLastSelectedLanguage(language);
+    this.setState({ language });
+  };
+
+  persistLastSelectedLanguage = language => {
+    if (window.localStorage) {
+      localStorage.setItem("lastSelectedLanguage", language);
```

Comment:
can we abstract out the local storage in a different file? going forward i see more integrations with local storage, and it would be useful to have that in one place versus scattered in the codebase

Reply:
addressed in https://github.com/covid-maps/covid-maps/pull/153/commits/2228fa322ac97cb1ffc08a4988647f7ab6a1421f and https://github.com/covid-maps/covid-maps/pull/153/commits/b2f163018bdc14c74665987ec7266d6bba4a1e61